src/main/java/org/testng/annotations/IConfigurationAnnotation.java

```diff
@@ -58,4 +58,12 @@
    *     a @BeforeSuite or similar that is represented as a configuration annotation.
    */
   boolean isFakeConfiguration();
+
+  default boolean isBeforeGroups() {
+    return false;
```

Comment:
Why do you use a default method here? Where is it missing?

Reply:
They aren't missing anywhere. This interface seems to be exposed (Since it doesn't reside within the `internal` package), I added them as `default` methods to ensure that we dont have backward compatibility issues.